nifi-provenance-reporting-tasks/src/main/java/io/egm/nifi/reporting/EmailProvenanceReporter.java

```diff
@@ -244,14 +265,41 @@ private Charset getCharset(final ReportingContext context) {
         return Charset.forName(context.getProperty(INPUT_CHARACTER_SET).getValue());
     }
 
-    private String composeMessageContent(final Map<String, Object> event) {
+    /**
+     * Utility function to get the nifi instance from the {NIFI_INSTANCE} property
+     *
+     * @param context the ProcessContext
+     * @return the Nifi Instance
+     */
+
+    private String getNifiInstance(final ReportingContext context) {
+        return context.getProperty(NIFI_INSTANCE).getValue();
+    }
+
+    private String getSpecificRecipientValue(final ReportingContext context, final Map<String, Object> event)  {
+        final String specificRecipientAttributeName = context.getProperty(SPECIFIC_RECIPIENT).getValue();
+        final String regexPattern = "^[A-Za-z0-9_-]+(\\.[A-Za-z0-9_-]+)*@[^-][A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)*(\\.[A-Za-z]{2,})$";
```

Comment:
the validity of the email address is already checked in the `toInetAdresses` function, it can be remove from this function